server/src/main/java/com/hedera/block/server/verification/session/BlockVerificationSessionFactory.java

```
@@ -0,0 +1,76 @@
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.server.verification.session;
+
+import com.hedera.block.server.metrics.MetricsService;
+import com.hedera.block.server.verification.VerificationConfig;
+import com.hedera.block.server.verification.signature.SignatureVerifier;
+import com.hedera.hapi.block.stream.output.BlockHeader;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.concurrent.ExecutorService;
+import javax.inject.Inject;
+
+/**
+ * A factory for creating block verification sessions.
+ */
+public class BlockVerificationSessionFactory {
+
+    private final VerificationConfig config;
+    private final MetricsService metricsService;
+    private final SignatureVerifier signatureVerifier;
+    private final ExecutorService executorService;
+    private final int hashCombineBatchSize;
+
+    /**
+     * Constructs a block verification session factory.
+     *
+     * @param verificationConfig the verification configuration
+     * @param metricsService the metrics service
+     * @param signatureVerifier the signature verifier
+     * @param executorService the executor service
+     */
+    @Inject
+    public BlockVerificationSessionFactory(
+            @NonNull final VerificationConfig verificationConfig,
+            @NonNull final MetricsService metricsService,
+            @NonNull final SignatureVerifier signatureVerifier,
+            @NonNull final ExecutorService executorService) {
+        this.config = verificationConfig;
+        this.metricsService = metricsService;
+        this.signatureVerifier = signatureVerifier;
+        this.executorService = executorService;
```

lets add require non null check on 52-55 as they are assignment only, lets fail early here

---

✅ Done!

---

You can combine the requireNonNull() checks with variable assignments here too for all these:
```
this.config = requireNonNull(verificationConfig);
```